src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/api/DocumentController.kt

```diff
@@ -0,0 +1,116 @@
+package org.vaccineimpact.orderlyweb.controllers.api
+
+import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.FileSystem
+import org.vaccineimpact.orderlyweb.Files
+import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
+import org.vaccineimpact.orderlyweb.db.Config
+import org.vaccineimpact.orderlyweb.db.DocumentRepository
+import org.vaccineimpact.orderlyweb.models.Document
+
+class DocumentController(context: ActionContext,
+                         private val fileSystem: FileSystem = Files(),
+                         private val config: Config = AppConfig(),
+                         private val repo: DocumentRepository): Controller(context)
+{
+    private val topLevelFolder = formatFolder(config["documents.location"])
+
+    fun refreshDocuments() {
+        val allDocs = repo.getAllFlat()
+
+        val unrefreshedDocs = allDocs.filter{it.isFile}.toMutableList()
+        val unrefreshedDirs = allDocs.filter{!it.isFile}.toMutableList()
+
+        val topLevelDir = unrefreshedDirs.find{it.path == relativePath(topLevelFolder)}
+        if ( topLevelDir == null)
+        {
+            repo.add(relativePath(topLevelFolder), documentName(topLevelFolder), false, null)
+        }
+        else
+        {
+            unrefreshedDirs.remove(topLevelDir)
+        }
+        refreshDocumentsInDir(topLevelFolder,  unrefreshedDocs, unrefreshedDirs)
+
+        //Hide all known files and folders which were not found
+        unrefreshedDocs.filter{ it.show }.forEach{ repo.setVisibility(it, false) }
+        unrefreshedDirs.filter{ it.show }.forEach{ repo.setVisibility(it, false) }
+    }
+
+    private fun refreshDocumentsInDir(absolutePath: String, //this needs to be absolute
+                                      unrefreshedDocs: MutableList<Document>,
+                                      unrefreshedDirs: MutableList<Document>)
+    {
+        val relativeParent = formatFolder(relativePath(absolutePath))
+
+        val files = fileSystem.getChildFiles(absolutePath).map{relativePath(it)}
```

Comment:
It feels like this method could be tidier if there was just one getChildren method which gave back both files and folders with some way of easily distinguishing them. It could give back a list of Files I guess, though this would be more of a pain to mock..